### DISCUSSION TITLE
IOS: timeout on main thread

### DISCUSSION THREAD

**Discussion by FM1973** [dotnet/maui#12382] [General] [answered]
Hey there.

Is it possible that there is a timeout on the main thread on IOS?
In one project we use ef core with migrations at startup. The migration takes some time and the app crashes after a couple of seconds.
This does only happen on a real device - not in the simulator, It does not happen on Android (neither simulator nor real device). But I can see that the thread the migration runs on, exits.
When we do the migration on a background thread, the app runs as expected.
So this isn´t a show stopper. I just wanted to know whether there is some sort of timeout or not.
Thanks!

**Comment by AnthonyIacono** [comment, marked as answer]
I know on iOS locking the main thread for an extended period of time can lead to the OS terminating your app, but not entirely sure about Android.

If you are in a callee of the FinishedLaunching method of AppDelegate, you get 17 seconds and if you do not return, the app is killed. This might be what is happening, especially if you are triggering these in "app launch" events.
